executor/simple.go

```diff
@@ -1556,7 +1557,22 @@ func (e *SimpleExec) executeShutdown(s *ast.ShutdownStmt) error {
 // This function need to run with async model, otherwise it will block main coroutine
 func asyncDelayShutdown(p *os.Process, delay time.Duration) {
 	time.Sleep(delay)
-	err := p.Kill()
+	// Send SIGTERM instead of SIGKILL to allow graceful shutdown and cleanups to work properly.
+	err := p.Signal(syscall.SIGTERM)
+	if err != nil {
+		panic(err)
+	}
+
+	// Sending SIGKILL should not be needed as SIGTERM should cause a graceful shutdown after
+	// n seconds as configured by the GracefulWaitBeforeShutdown. This is here in case that doesn't
+	// work for some reason.
+	graceTime := config.GetGlobalConfig().GracefulWaitBeforeShutdown
+	if graceTime < 5 {
+		graceTime = 5
+	}
+	time.Sleep(time.Second * time.Duration(graceTime*2))
```

Comment:
Why wait double the grace period? I think it is more predictable to wait exactly the graceTime.

Reply:
When this works as it should the real shutdown starts after the grace period. I don't want to kill the process while the shutdown is happening. So either we wait for `graceTime+30` or for `graceTime*2` or so.